### PR TITLE
Constants to db

### PR DESCRIPTION
**Pull request by ArturFedorov**

### CODE REVIEW

File: client/src/components/projects/project-filter/ProjectFilter.vue [hunk 1]
```diff
@@ -4,13 +4,13 @@
       :name="model.name"
       :opened="model.opened"
       @update:opened="model.opened = !model.opened"
-      v-for="model in models">
+      v-for="model in modelsComputed">
```
Comment (Igogrek) [inline client/src/components/projects/project-filter/ProjectFilter.vue]: Do not name properties with computed prefix - just use models

File: client/src/components/projects/project-filter/ProjectFilter.vue [hunk 2]
```diff
@@ -23,36 +23,39 @@
...
-
+  import * as types from '../../../store/modules/projects/project-types';
+  import {IModel} from '../../../shared/interfaces/IModel';
+  import {IProgram} from "../../../shared/interfaces/IProgram";
 
   export default Vue.extend({
       data() {
         return {
-          models: new Array<IProjectFilter>()
+          models: [] as IProjectFilter[]
```
Comment (Igogrek) [inline client/src/components/projects/project-filter/ProjectFilter.vue]: Does this data still needed?

File: client/src/components/projects/project-filter/ProjectFilter.vue [hunk 3]
```diff
@@ -23,36 +23,39 @@
...
-          };
+        this.$store.dispatch(types.FETCH_ADDONS);
+      },
+      computed: {
 
-          return model;
-        });
+        modelsComputed(): IProjectFilter[] {
+          const addons = this.$store.getters[types.GET_ADDONS];
+          this.models = [] as IProjectFilter[];
```
Comment (Igogrek) [inline client/src/components/projects/project-filter/ProjectFilter.vue]: Do you really need to use data field here? could data be removed at all ?

File: client/src/components/projects/project-filter/ProjectFilter.vue [hunk 4]
```diff
@@ -23,36 +23,39 @@
...
+            const model: IProjectFilter = {
+              name: key,
+              opened: true,
+              items: this.createModelForCheckboxes(addons[key])
+            };
 
+            this.models.push(model);
+          });
+
+          return this.models;
```
Comment (Igogrek) [inline client/src/components/projects/project-filter/ProjectFilter.vue]: Again - remove data and just return some array

File: client/src/store/modules/projects/getters.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+export const getters = {
+
+  /**
+   * Get properties only for
+   * project filter
+   * @param {IProjectState} state
+   * @returns {{}}
+   */
+  [GET_ADDONS](state: IProjectState) {
+    return {
```
Comment (Igogrek) [inline client/src/store/modules/projects/getters.ts]: Not sure if this getter is right - usually getters are functions like:
[Types.PRODUCTION_LINE]: state => state.lines

File: client/src/store/modules/projects/mutations.ts [hunk 6]
```diff
@@ -1,13 +1,34 @@
...
 
 export const mutations = {
   [SET_PROJECTS](state: IProjectState, payload: IProject[]) {
     state.projects = payload;
     state.loading = false;
   },
-  [SET_CUSTOMERS](state: IProjectState, payload: String[]) {
+  [SET_LINES](state: IProjectState, payload: ILine[]) {
+    state.lines = payload;
+    state.loading = false;
```
Comment (Igogrek) [inline client/src/store/modules/projects/mutations.ts]: Extract loading to separate mutation and just commit after all SET_'s

File: server/features/project/shared/project-addons.service.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+const projectAddonsService = {
+
+  getProjectFilterModel: async () => {
+    const lines = await Line.scope([Scopes.WITH_PROGRAMS]).findAll();
+    const programs = await Program.scope([Scopes.WITH_LINE]).findAll({ order: [['name', 'ASC']] });
+    const types = await Type.findAll();
+    const domains = await Domain.findAll({ order: [['name', 'ASC']] });
+    const customers = await  Customer.findAll({ order: [['name', 'ASC']] });
+
+    const dto = new ProjectFilterDto();
```
Comment (Igogrek) [inline server/features/project/shared/project-addons.service.ts]: add constructor to ProjectFilterDto and do like

new ProjectFilterDto(lines, programs, types)

File: client/src/store/modules/projects/index.ts [hunk 8]
```diff
@@ -1,21 +1,36 @@
...
+  domains: IDomain[];
+  types: IType[];
+  customers: ICustomer[];
   loading: boolean;
 }
 
 const initialState: IProjectState = {
-  projects: [],
-  customers: [],
+  projects: new Array<IProject>(),
```
Comment (Igogrek) [inline client/src/store/modules/projects/index.ts]: Do not do this, just do projects: []

File: client/src/components/projects/project-filter/ProjectFilter.vue (hunk 4 above)
Comment (ArturFedorov) [reply client/src/components/projects/project-filter/ProjectFilter.vue]: doesn't work without data, checkboxes won't update accordion won't toggle

File: client/src/store/modules/projects/index.ts (hunk 8 above)
Comment (ArturFedorov) [reply client/src/store/modules/projects/index.ts]: done

File: server/features/project/shared/project-addons.service.ts (hunk 7 above)
Comment (ArturFedorov) [reply server/features/project/shared/project-addons.service.ts]: done

File: client/src/store/modules/projects/mutations.ts (hunk 6 above)
Comment (ArturFedorov) [reply client/src/store/modules/projects/mutations.ts]: done

File: client/src/store/modules/projects/getters.ts (hunk 5 above)
Comment (ArturFedorov) [reply client/src/store/modules/projects/getters.ts]: Idea behind, Types.PRODUCTION_LINE - headers of accordion, lines/ domains ets are checkboxes values, getter in order to iterate among fewer number of keys (here we have no projects and loading)

**Pull request merged**